### DISCUSSION TITLE
Coordinates in yscene

### DISCUSSION THREAD

**Discussion by M1ndHunteR** [xelatihy/yocto-gl#1234] [Q&A] [answered]
Hello!

I'm trying to perform some basic rendering experiments using yscene and json file defining a scene.
I am having a great deal of problem to understand the coordinate systems used in "lookat" and "frame" for the camera and the object of interest. I assume I can use these to position and direct both the camera and the object....
I couldn't find any info on in the documentation. Anyone who has some info, please shed some light on this stuff. 

Kind regards,
Eugene.

**Comment by giacomonazzaro** [comment, marked as answer]
The "lookat" syntax for expressing frames in the json format matches the arguments of the function `lookat_frame()` in `libs/yocto/yocto_math.h`, with vector values unpacked.

Let's say you want to set a camera that, from point `(1, 2, 3)`, looks at point `(0, 0, 0)`, with up vector `(0, 1, 0)`.
In the json format you will write:

```
...
"cameras": {
  "my_camera": {
    "lookat": [1,2,3,  0,0,0,  0,1,0],
    "lens": 0.1,
    "aperture": 0.0,
    "aspect": 2.4
  }
},
...
```

**Reply by M1ndHunteR** [reply]
Thanks for the answer. It helps.
Would you be so kind to also explain me what is the "frame" in "shape" settings?
Greatly appreciated!
Eugene

**Reply by xelatihy** [reply]
In instances, the frame is an affine transform written in column-major format.
So if you want to place an object at `(1,2,3)` you can write

```
"frame": [1,0,0,   0,1,0,   0,0,1,   1,2,3]
```

Rotations are hard to write this way, so you can use the lookout on instances too.
